src/optimizer/rule/IndexScanRule.cpp

```diff
@@ -402,6 +430,16 @@ IndexItem IndexScanRule::findOptimalIndex(graph::QueryContext *qctx,
     return indexesRange[0];
 }
 
+// Only use "lookup on tagname"
+IndexItem IndexScanRule::findOptimalIndex(graph::QueryContext *qctx,
+                                          const OptGroupNode *groupNode) const {
+    auto indexes = allIndexesBySchema(qctx, groupNode);
+    if (indexes.empty()) {
+        return nullptr;
+    }
+    return indexes[0];
+}
+
```

Comment:
A suggestion . rename the method to `findLightestIndex`, and find the lightest index. for example : 
index1 on tag ()
index2 on tag(c1)
index3 on tag(c1,c2,...)
Index1 should be selected. right?

Reply:
The current strategy is to find an index randomly.

Reply:
Here, the first index is used as the index to be used for returning.

Reply:
> Here, the first index is used as the index to be used for returning.

I think, the best scanning performance of the storage layer is the lightest index. WDYT?
In addition, if return the first index, this method name is best `getFirstIndex` or `findAnyIndex` . right?

Reply:
Make sense,  Let’s use the index with the fewest fields, 👍

Reply:
Fixed.